rest_tools/client/device_client.py

```diff
@@ -131,7 +119,85 @@ def DeviceGrantAuth(
             raise RuntimeError('Device authorization failed') from exc
         break
 
-    refresh_token = req['refresh_token']
+    return req['refresh_token']
+
+
+def DeviceGrantAuth(
```

Comment:
Is there any reason why you've chosen to use pascal case for these functions?

Reply:
I think it was so they look to the user identical to the other Client classes, even if they're more of a factory function into them.  No great reason though.

Reply:
I don't dislike it entirely. Devs won't be using the returned instance much, just set and forget. The alternative is to use class inheritance--which might be more convoluted? Not sure

Reply:
Class inheritance is definitely more convoluted, but could be done.  Mostly it was that I could do a few starter actions, then pass to the existing class.  You could also do something like:
```python
class A(B):
  def __init__(self, arg1, arg2):
    # do startup things
    super().__init__(arg2)
```
but that seems a bit messy.